src/index.ts

```diff
@@ -26,58 +23,51 @@ type CLIParams = {
-  const traceSamples = await api.getTraceSamples({
-    ...params,
-    transactionName,
-    transactionType,
-    service,
-  });
-
-  const uniqueTraceSamples = _.uniqBy(traceSamples, 'traceId')
-
-  const traces = await bluebird.map<_, Events[]>(uniqueTraceSamples, ({ traceId }) => api.getTrace({
-    id: traceId,
-    start: params.start,
-    end: params.end
-  }), { concurrency: 10 });
-
-
-  const rawTraces = traces.map(events => tree(events.map(sanitize.scalability))).filter(e => !_.isEmpty(e))
-
-  return {
-    service,
-    transactionType,
-    transactionName,
-    traces: rawTraces
-  }
-};
-
 const apmParser = async ({ param, client }: CLIParams) => {
-  const api = new ApmClient(client);
-  console.log(`Querying APM with condition (${param.kuery}) starting from ${param.start} and ending at ${param.end}`);
-
-  const transactionGroupMainStatistics = await Promise.all(param.transactionType.map(transactionType => api.getTransactionGroupMainStatistics(transactionType === 'http-request' ? 'kibana-frontend' : 'kibana', {
-    latencyAggregationType: 'avg',
-    transactionType,
-    ..._.pick(param, ['start', 'end', 'kuery', 'environment'])
-  })))
-
-  const response = await bluebird.mapSeries(transactionGroupMainStatistics.flat(), i => {
-    const service = i.transactionType === 'http-request' ? 'kibana-frontend' : 'kibana'
-    return enrichTrace(i.name, i.transactionType, service, api, param)
+  const authOptions = {
+    node: client.baseURL,
+    username: client.auth.username,
+    password: client.auth.password,
+  }
+  const esClient = initClient(authOptions);
+  const hits = await esClient.getTransactions(param.jobId, param.start);
+  if (!hits && hits.length === 0) {
+    throw new Error('No transactions found')
+  }
+  const data = hits.map(hit => hit._source).map(hit => {
+    return  {
+      processor: hit.processor,
+      labels: hit.labels,
+      traceId : hit.trace.id,
+      timestamp: hit["@timestamp"],
+      environment: hit.environment,
+      request: {
+        url: { path: hit.url.path },
+        headers: hit.http.request.headers,
+        method: hit.http.request.method,
+        body: hit.http.request.body ? JSON.parse(hit.http.request.body.original) : '',
+      },
+      response: {statusCode: hit.http.response.status_code},
+      transaction: {
+        id: hit.transaction.id,
+        name: hit.transaction.name,
+        type: hit.transaction.type,
+      }
+    }
   });
 
+  console.log(`Found ${hits.length} hits`);
+
   const output = {
     journeyName: "Sample Journey Name",
     journeyTime: 300000,
     kibanaVersion: "v7.17.1",
     kibanaUrl: "https://kibana-ops-e2e-perf.kb.us-central1.gcp.cloud.es.io/",
     maxUsersCount: 1,
-    traceItems: response,
+    traceItems: data,
   }
 
-  await writeFile(`trace.json`, JSON.stringify(output));
+  await writeFile(`trace.json`, JSON.stringify(output, null, 2), 'utf8');
```

Comment:
I think it's not super important right now we might wanna use job id only as a name. Since we are going to create the artifact after SUPT it will be based on the environment variable we get from BUILDKITE. I can modify these parts when we publish apm-parser as a package and integrate into CI

Reply:
I will use `journeyName` value instead since we expect it to be unique within run